src/main/java/com/google/api/codegen/gapic/MainGapicProviderFactory.java

```diff
@@ -235,10 +220,27 @@
               .setCodePathMapper(pythonPathMapper)
               .build();
 
+      if (id.equals(PYTHON)) {
+        System.out.println("stopping here");
+        return Arrays.<GapicProvider<? extends Object>>asList(mainProvider, clientConfigProvider);
+      }
+
+      System.out.println("not stopping");
+      GapicProvider<? extends Object> messageProvider =
+          CommonGapicProvider.<ProtoFile>newBuilder()
+              .setModel(model)
+              .setView(new ProtoFileView())
+              .setContext(new PythonGapicContext(model, apiConfig))
+              .setSnippetSetRunner(
+                  new PythonSnippetSetRunner<ProtoFile>(
+                      new PythonProtoFileInitializer(), SnippetSetRunner.SNIPPET_RESOURCE_ROOT))
+              .setSnippetFileNames(Arrays.asList("py/message.snip"))
+              .setCodePathMapper(CommonGapicCodePathMapper.defaultInstance())
+              .build();
       return Arrays.<GapicProvider<? extends Object>>asList(
           mainProvider, messageProvider, clientConfigProvider);
 
-    } else if (id.equals(RUBY)) {
+    } else if (id.equals(RUBY) || id.equals(RUBY_DOC)) {
```

Comment:
Yes, "doc" is a superset. It includes an additional set of files that are not functional but that can be used to generate docs. For Python this is problematic because these files actually shadow real, useful files and break the client.